src/main/java/io/stargate/sgv2/jsonapi/exception/JsonApiException.java

```diff
@@ -23,6 +23,8 @@ public class JsonApiException extends RuntimeException implements Supplier<Comma
 
   private final ErrorCode errorCode;
 
+  private final Response.Status httpStatus;
```

Comment:
Will try propagating HTTP response code along with the exception so it's neither lost nor need to be carried separately.